src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/InstancesController.cs

```diff
@@ -121,20 +249,26 @@ public async Task<ActionResult> Get(int instanceOwnerId, Guid instanceGuid)
         /// <returns>instance object</returns>
         /// <!-- POST /instances?appId={appId}&instanceOwnerId={instanceOwnerId} -->
         [HttpPost]
-        public async Task<ActionResult> Post(string appId, int instanceOwnerId, [FromBody] Instance instanceTemplate)
+        public async Task<ActionResult> Post(string appId, int? instanceOwnerId, [FromBody] Instance instanceTemplate)
         {
-            if (instanceTemplate == null && instanceOwnerId == 0)
+            if (instanceTemplate == null && !instanceOwnerId.HasValue)
             {
                 return BadRequest("Missing parameter values: instanceOwnerId must be set");
             }
-            else if (instanceOwnerId == 0 && (instanceTemplate != null && string.IsNullOrEmpty(instanceTemplate.InstanceOwnerId))) 
+            else if (!instanceOwnerId.HasValue && (instanceTemplate != null && string.IsNullOrEmpty(instanceTemplate.InstanceOwnerId))) 
             {
                 return BadRequest("Missing parameter values: instanceOwnerId must be set");
             }
 
-            if (instanceOwnerId == 0 && instanceTemplate != null)
+            string theInstanceOwnerId = null;
```

Comment:
Could a defaultvalue for the input parameter be a better solution here? 
that way we can avoid the theInstanceOwnerId parameter name.

Reply:
I have refactored this in another branch.